service/frontend/handler.go

```diff
@@ -558,11 +558,7 @@ func (wh *WorkflowHandler) GetWorkflowExecutionHistory(
 		return nil, errWorkflowIDNotSet
 	}
 
-	if !getRequest.GetExecution().IsSetRunId() {
-		return nil, errRunIDNotSet
-	}
-
-	if uuid.Parse(getRequest.GetExecution().GetRunId()) == nil {
+	if getRequest.GetExecution().IsSetRunId() && uuid.Parse(getRequest.GetExecution().GetRunId()) == nil {
```

Comment:
It is intentional, to allow getting history of running execution using the workflow_id only